File: ios/ZebraStripedModule.swift
import ExpoModulesCore

public class ZebraStripedModule: Module {
    public func definition() -> ModuleDefinition {
        Name("ZebraStriped")
        
        Function("getBase64Code") { (value: String, format: String, size: CGSize, onColor: CGColor?, offColor: CGColor?) in
            let barcodeFormat = ZebraFactory.parseToBarcodeFormat(codeFormat: CodeFormat(rawValue: format)!)
            
            return ZebraFactory.generateCodeImage(value: value, format: barcodeFormat, size: size, onColor: onColor, offColor: offColor).pngData()?.base64EncodedString()
        }
        
        View(ZebraCodeView.self) {
            Prop("value") { (view: ZebraCodeView, value: String) in
                view.value = value
            }
            
            Prop("format") { (view: ZebraCodeView, format: String) in
                view.updateFormat(CodeFormat(rawValue: format)!)
            }
            
            Prop("size") { (view: ZebraCodeView, size: CGSize) in
                view.size = size
            }
            
            Prop("onColor") { (view: ZebraCodeView, color: CGColor?) in
                view.onColor = color
            }
            
            Prop("offColor") { (view: ZebraCodeView, color: CGColor?) in
                view.offColor = color
            }
        }
    }
}

File: android/src/main/java/expo/modules/zebrastriped/ZebraStripedModule.kt
package expo.modules.zebrastriped

import android.graphics.Bitmap
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.io.ByteArrayOutputStream
import java.lang.Exception
import java.util.Base64

class ZebraStripedModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ZebraStriped")

    Function("getBase64Code") { value: String, format: String, size: Pair<Int, Int>, onColor: String?, offColor: String? ->
      val factory = ZebraFactory()
      val barcodeFormat = factory.parseToBarcodeFormat(CodeFormat.valueFromString(format))

      val bos = ByteArrayOutputStream()
      val bitmap = factory.generateCodeBitmap(value, barcodeFormat, size, onColor, offColor)
      bitmap?.compress(Bitmap.CompressFormat.PNG, 100, bos);

      return@Function Base64.getEncoder().encodeToString(bos.toByteArray())
    }

    View(ZebraCodeView::class) {
      Prop("value") { view: ZebraCodeView, value: String ->
        view.value = value
      }

      Prop("format") { view: ZebraCodeView, formatString: String ->
        view.updateFormat(CodeFormat.valueFromString(formatString))
      }

      Prop("size") { view: ZebraCodeView, size: Pair<Int, Int> ->
        view.size = size
      }

      Prop("onColor") { view: ZebraCodeView, color: String ->
        view.onColor = color
      }

      Prop("offColor") { view: ZebraCodeView, color: String ->
        view.offColor = color
      }
    }
  }
}
